truss/truss_config.py

```diff
@@ -139,6 +143,74 @@ def to_dict(self):
         }
 
 
+@dataclass
+class ExternalDataItem:
+    """A piece of remote data, to be made available to the Truss at serving time.
+
+    Remote data is downloaded and stored under Truss's data directory. Care should be taken
+    to avoid conflicts. This will get precedence if there's overlap.
+    """
+
+    # Url to download the data from.
+    # Currently only files are allowed.
+    url: str
+
+    # This should be relative path. This is where the remote file will be downloaded.
+    at: str
+
+    # The backend used to download files
+    backend: str = DEFAULT_BLOB_BACKEND
+
+    # A name can be given to a data item for readability purposes. It's not used
+    # in the download process.
+    name: Optional[str] = None
+
+    @staticmethod
+    def from_dict(d: Dict[str, str]) -> "ExternalDataItem":
```

Comment:
[absolutely non-blocking] I don't know if we've discussed this in the past -- have we considered using pydantic instead of dataclasses for cases like this? You get built-in validations / runtime type checking so we wouldn't have to do these types of checks ourselves.

This might be a bigger change, and I know we probably want to be cautious about adding new dependencies to Truss, but wanted to throw this out there.

Reply:
Great suggestion. Yes, I think the config is getting complex enough that it's worth looking into that. It seems that pydantic is really lean:
```
(bin) @pankajroark ➜ /workspaces/foo $ pip show pydantic
Name: pydantic
Version: 1.10.7
Summary: Data validation and settings management using python type hints
Home-page: https://github.com/pydantic/pydantic
Author: Samuel Colvin
Author-email: s@muelcolvin.com
License: MIT
Location: /workspaces/foo/bin/lib/python3.9/site-packages
Requires: typing-extensions
Required-by: 
```

I'll look into moving to pydantic as a follow-up.